### ISSUE TITLE
[BUG] - Wrong indexing causes Memgraph to become unresponsive and forces you to restart whole instance 

### ISSUE THREAD

**Issue by antejavor** [memgraph/memgraph#1415]
Reporting this based on the context from [discord] (https://discord.com/channels/842007348272169002/890968055339958292/1144023890528575528). You can read the full context there; below is the discussion summary. 

**Memgraph version**
2.11, AWS docker 

**Describe the bug**
If you start running Memgraph and loading larger size of CSV files (500k, 1M or more) that are creating edges between nodes on a sufficient scale, and at that point, if you have misconfigured the index and matching a node without an index (probably ScanALL) then Memgraph will be pegged at 2-3 edges per seconds. 

During the process, Memgraph will become unresponsive to the clients (web, lab, driver) and eat all CPU resources, until import is completed and it chills out. This forces users to restart their instances during the process, which opens another set of problems regarding data loss, etc. 

Even worse, it makes people think that Memgraph has died, by closer inspection, it turns out Memgraph is just using 100% CPU resources and moves itself to an unresponsive state. 

**To Reproduce**
Data + cypher file: https://drive.google.com/drive/folders/1hd3cTZw13k1hAPQoeVwNIbhlEzp1R25C?usp=sharing

**Expected behavior**
Memgraph should enable clients to open sessions and connect to Memgraph even if Memgraph is doing the import, it should not overrun all resources to stop users from working. This does not give you the option to stop the [transactions you have previously started ](https://memgraph.com/docs/fundamentals/transactions#terminate-transactions).

**Comment by antejavor** [comment]
I think this issue may be troublesome for many folks who do not configure indexes properly. Indexes are quite critical in GraphDB and Memgraph, and we often see that mistake. If you do this on a bigger scale, it will make Memgraph unresponsive and pegged, which makes people think Memgraph has died by closer inspection, it turns out Memgraph is just using 100% CPU resources and moves itself to an unresponsive state.  Hence the appropriate I2 and S2 tags.

**Comment by antejavor** [comment]
Related issue and potential solution: https://github.com/memgraph/memgraph/issues/1084

**Comment by as51340** [comment]
Few things going on here:
1. there is a wrong setup in the cypher file the user is using for the import because there is no index on `Org(name)` which is used in multiple occasions for importing edges. Without using indices for importing edges, the speed will be very bad unfortunately and that is completely expected and normal since indices reduce you the complexity of the search.
2. I am not sure how the user does the import since he didn't specify but if he is doing sequential import in a way that he runs LOAD CSV commands one by one (most probably) then he should be able to open new client and terminate transaction which is causing him problem. I tested that multiple times and killing load csv command works as expected.
3. if he is doing parallel import in a way that he is doing multiple load csv commands in parallel then it is maybe possible that he cannot open a new session and terminate transaction. That would be possible if all cores are busy doing their things. However, I don't believe this is the case here.

**Comment by as51340** [comment]
Can you please verify @antejavor that import of edges has expected speed once label-property index `Org(name)` will be created?

**Comment by antejavor** [comment]
Yes, once indexes are properly set, there is no issue. This issue deals with problem 3,  he did mention in the discord thread that CPU was 100% importing edged 2-3 edges per second. Not sure it was a parallel import, but it sounds like it should be. I will double-check.

**Comment by as51340** [comment]
please double-check parallel import yes because I don't think he uses parallel import. I think it's normal that CPU is 100% while importing 2-3 edges without indexes. I would expect that.

**Comment by antejavor** [comment]
Okay, so there are two things here: 

- We need to see in what exact scenario Memgraph becomes unresponsive, @as51340 was not able to replicate this while loading the CSV file. 
- We need to see if this is related to the indexing or not.

**Comment by antejavor** [comment]
> and running a CREATE SNAPSHOT after every 1m entries (since it tends to die every 2M or so).

@as51340 On the discord thread, the user also provided this info, on a bit larger scale that could cause Memgraph to get stuck?
